src/index.jsx

```diff
@@ -29,7 +29,7 @@ subscribe(APP_READY, () => {
           <Routes>
             <Route
               path={ROUTES.PROGRAM_RECORDS}
-              element={<ProgramRecordsList />}
+              element={<AuthenticatedPageRoute><ProgramRecordsList /></AuthenticatedPageRoute>}
```

Comment:
I was under the impressions that the `AuthenticatedPageRoute` component was actually a react-router `Route` of its own. This is indicated [in the docs](https://github.com/openedx/frontend-platform/blob/93ea5b25423e8b607fb1c8633778503d76f47509/src/react/AuthenticatedPageRoute.jsx#L9-L12), HOWEVER, the implementation used in `frontend-platform` test files just [uses it as a wrapper](https://github.com/openedx/frontend-platform/blob/93ea5b25423e8b607fb1c8633778503d76f47509/src/react/AuthenticatedPageRoute.test.jsx#L44) (like this PR does).

When digging into the code, I don't see anywhere that `AuthenticatedPageRoute` would ever return an actual `Route` component. See the [code in frontend-platform](https://github.com/openedx/frontend-platform/blob/93ea5b25423e8b607fb1c8633778503d76f47509/src/react/AuthenticatedPageRoute.jsx#L35-L37).

What's strange is that there ARE implementations where MFEs are adding props like `path` and `component` to the `AuthenticatedPageRoute` component. See [here in frontend-app-learner-portal-enterprise](https://github.com/openedx/frontend-app-learner-portal-enterprise/blob/451b94b85eeef96e7527968ecfe383fab58e71ff/src/components/app/App.jsx#L55-L56).

All of this to say, this PR is clearly fine but I'm also clearly a little confused so will be bringing this to fedX group to hopefully get some clarification on how this component is intended to be used.

Reply:
Hey thanks for bringing in all of these PR examples! Very valid point in terms of implementation. Originally I did try replacing the `Route` element with a `AuthenticatedPageRoute`, but then an error appeared: 

`[AuthenticatedPageRoute] is not a <Route> component. All component children of <Routes> must be a <Route> or <React.Fragment>`

This must be a hard requirement for frontend-platform v5, and previous PRs might need to be updated to reflect this rule.

Reply:
One other thing to note is that the comments[ in the docs](https://github.com/openedx/frontend-platform/blob/93ea5b25423e8b607fb1c8633778503d76f47509/src/react/AuthenticatedPageRoute.jsx#L9-L12) is from 4 years ago (found via git blame), so it should be updated to better represent its implementation.

Reply:
And lastly, it looks like the implementation in `frontend-app-learner-portal-enterprise` has placed the routes directly into `App.jsx` (and not wrapped with a `Routes` element), and this was done[ over two years ago](https://github.com/openedx/frontend-app-learner-portal-enterprise/blame/9b70f62e2df63754bd8c592c80e8498b6df4eb09/src/components/app/App.jsx#L30). And they're still using `frontend-platform v4`.

So from that, I think that `AuthenticatedPageRoute` once had served as a substitute for `Route`, but it no longer does in v5.